.gitlab-ci.yml

```diff
@@ -0,0 +1,268 @@
+    # @TODO: Don't use -j on Gitlab.com as builds just get stuck when running
+    # multi-proc, needs more debugging
+    - make test || true  # Unit tests constantly fail, see https://jira.mariadb.org/browse/MDEV-25820
+    # - make test-force || true  # mysql-test-runner takes too long, run it in a separate job instead
+    - *rpm_listfiles
+    - mkdir ../rpm; mv *.rpm ../rpm
+    - ccache -s
+  artifacts:
+    when: always  # Must be able to see logs
+    paths:
+      - build-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+      - rpmlist-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+      - rpm
+      - builddir/_CPack_Packages/Linux/RPM/SPECS/
+  cache:
+    key: $MARIADB_MAJOR_VERSION-$CI_JOB_NAME
+    paths:
+      - .ccache
+
+centos7:
+  stage: build
+  image: centos:7
+  variables:
+    GIT_STRATEGY: fetch
+    GIT_SUBMODULE_STRATEGY: normal
+  script:
+    - yum install -y yum-utils rpm-build gcc gcc-c++ bison libxml2-devel libevent-devel openssl-devel
+    - yum-builddep -y mariadb-server
+    - mkdir builddir; cd builddir
+    - cmake -DRPM=$CI_JOB_NAME $CMAKE_FLAGS .. 2>&1 | tee -a ../build-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+    - make package 2>&1 | tee -a ../build-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+    # @TODO: Don't use -j on Gitlab.com as builds just get stuck when running
+    # multi-proc, needs more debugging
+    - make test || true  # Unit tests constantly fail, see https://jira.mariadb.org/browse/MDEV-25820
+    # - make test-force || true  # mysql-test-runner takes too long, run it in a separate job instead
+    - *rpm_listfiles
+    - mkdir ../rpm; mv *.rpm ../rpm
+  artifacts:
+    when: always  # Must be able to see logs
+    paths:
+      - build-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+      - rpmlist-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+      - rpm
+      - builddir/_CPack_Packages/Linux/RPM/SPECS/
+
+mysql-test-run:
+  stage: test
+  dependencies:
+    - fedora
+  script:
+    # Install packages so tests and the dependencies install
+    # @TODO: RPM missing 'patch' as dependency, so installing it manually for now
+    - yum install -y rpm/*.rpm patch
+    # @TODO: Fix on packaging level that /usr/share/mariadb would work and errormsg.sys be found
+    - rm -rf /usr/share/mariadb; ln -s /usr/share/mysql /usr/share/mariadb
+    # mtr expects to be launched in-place and with write access to it's own directories
+    # @TODO: debug test environment
+    - nproc; cat /proc/cpuinfo | grep -F "model name" -m 1; head -n 1 /proc/meminfo; df -h
+    - whoami; id;
+    - cd /usr/share/mysql-test
+    - |
+      echo "
+      main.mysqldump : flaky on Gitlab-CI
+      main.flush_logs_not_windows : flaky in containers in general
+      main.mysql_upgrade_noengine : requires diff but diffutils is not a dependency
+      " > skiplist
+    # @TODO: Flaky tests are skipped for now, but should be fixed
+    - ./mtr --suite=main --force --xml-report=$CI_PROJECT_DIR/junit.xml --skip-test-list=skiplist
```

Comment:
I see that this invocation is missing `--parallel=auto`. Do we really want to avoid running multiple concurrent instances of the server?
I think that it could be useful to additionally run all tests that are in `mysql-test/collections/smoke_test` that was recently introduced in 3635280cf7ef74ec18cfb21b21e779725aab186d.
For the smoke test, something like this could work:
```sh
./mtr $(cat …/mysql-test/collections/smoke_test) --verbose-restart --mem --parallel=auto --force --max-save-core=0 --max-save-datadir=1
```
You can find a `mtr-smoke-test` step on buildbot.askmonty.org.

Reply:
Why not add a `make smoke-test` target in the Makefile, like we already have `make test` and `make test-force`?

Reply:
Yes adding a `smoke-test` target (in a separate change) sounds like a good idea, to avoid duplicating logic.
Because the use of this script is voluntary, the lack of parallel compilation or testing is not a showstopper.